lib/modules/versioning/pvp/index.spec.ts

```diff
@@ -0,0 +1,351 @@
+import pvp, { extractAllComponents, getComponents, parse } from '.';
+
+describe('modules/versioning/pvp/index', () => {
+  describe('.isGreaterThan(version, other)', () => {
+    it('should return true', () => {
+      expect(pvp.isGreaterThan('1.23.1', '1.9.6')).toBeTrue();
+      expect(pvp.isGreaterThan('4.0.0', '3.0.0')).toBeTrue();
+      expect(pvp.isGreaterThan('3.0.1', '3.0.0')).toBeTrue();
+      expect(pvp.isGreaterThan('4.10', '4.1')).toBeTrue();
+      expect(pvp.isGreaterThan('1.0.0', '1.0')).toBeTrue();
+    });
+
+    it('should return false', () => {
+      expect(pvp.isGreaterThan('2.0.2', '3.1.0')).toBeFalse(); // less
+      expect(pvp.isGreaterThan('3.0.0', '3.0.0')).toBeFalse(); // equal
+      expect(pvp.isGreaterThan('4.1', '4.10')).toBeFalse();
+      expect(pvp.isGreaterThan('1.0', '1.0.0')).toBeFalse();
+    });
+  });
+
+  describe('.parse(range)', () => {
+    it('should parse >=1.0 && <1.1', () => {
+      const parsed = parse('>=1.0 && <1.1');
+      expect(parsed).not.toBeNull();
+      expect(parsed!.lower).toBe('1.0');
+      expect(parsed!.upper).toBe('1.1');
+    });
+  });
```

Comment:
What about ranges like `>=1.0`?

Reply:
The reason that I, in [the requirements discussion](https://github.com/renovatebot/renovate/discussions/31663), focused on this specific range format, is that it is pretty widely used. One way to gauge the frequency of use, is to look at the [Stackage incompatibility listings](https://github.com/commercialhaskell/stackage/issues/7548). As you can see, almost all of the bounds use the format implemented in this PR.

But you are of course correct that `>=1.0` is technically a valid version range. One reason that it isn't used much in the ecosystem, is likely because 'future' major releases (e.g. `1.1` and higher) are potentially incompatible, so it seems incorrect to claim compatibility with all future versions.

I'd *would* however, like to support arbitrary expressions. But I was imagining making it part of a follow-up expansion of this module. I have a draft, but it's hard to do elegantly, especially because I have to figure out how to support `getNewVersion` for the maximum subset of expressions, while retaining the style of expression. That is, if the user has `^>= 4.18 || ^>= 4.19`, they want Renovate to change it to `^>=4.18 || ^>=4.19 || ^>=4.20`. But if they use the style implemented in this PR, they don't want Renovate to just append `||^>=4.20`, as it would be weird (and IMHO unaccceptable) to have Renovate suggest `>= 4.18 && <4.20 || ^>=4.20`. I was imagining discussing this expansion *after* implementing datasource and manager. How does that sound?

Reply:
Ok, then this should documented in the readme and the modules ( in this case versioning ) marked as experimental.